### PR TITLE
SPEC-1060 Don't normalize username in SCRAM-SHA-256

### PR DESCRIPTION
**Pull request by xdg**

### CODE REVIEW

File: source/auth/auth.rst [hunk 1]
```diff
@@ -763,23 +764,29 @@ database command error.)
...
 ------
 
-To test SASLprep behavior, create a user with username and password equal to
-the Unicode character ROMAN NUMERAL NINE.  To create the user, use the
-(post-SASLprep) username and password "IX" and specify SCRAM-SHA-256
-credentials:
+To test SASLprep behavior, create two users:
+
+#. username: "IX", password "IX"
+#. username: "\u2168" (ROMAN NUMERAL NINE), password "\u2163" (ROMAN NUMERAL FOUR)
```
Comment (ajdavis) [inline source/auth/auth.rst]: Now that we're prohibiting SASLPrep for usernames, let's choose a non-ASCII username that is *not* equivalent to the ASCII one.

File: source/auth/auth.rst [hunk 2]
```diff
@@ -622,7 +622,8 @@ The MongoDB SCRAM-SHA-256 mechanism works similarly to the SCRAM-SHA-1
 mechanism, with the following changes:
 
 - The SCRAM algorithm MUST use SHA-256 as the hash function instead of SHA-1.
-- User names MUST be prepared with SASLprep, per RFC 5802.
+- User names MUST NOT be prepared with SASLprep.  This intentionally
+  contravenes the "SHOULD" provision of RFC 5802.
```
Comment (ajdavis) [inline source/auth/auth.rst]: Let's explain the reason for this in the Q&A please.

File: source/auth/auth.rst (hunk 1 above)
Comment (xdg) [reply source/auth/auth.rst]: It's intentional.  If an implementation does SASLprep when it shouldn't, "\u2168" will map to "IX" but have the wrong password so login will fail.

File: source/auth/auth.rst (hunk 2 above)
Comment (xdg) [reply source/auth/auth.rst]: Will do.

**Comment by xdg** [comment]
Rebased and merged as 8d3b2ef694c70833b4fa125910c8c4305dc0c7fd

File: source/auth/auth.rst (hunk 1 above)
Comment (ajdavis) [reply source/auth/auth.rst]: Oh, the password is different. I missed that NINE is not FOUR. LGTM.

File: source/auth/auth.rst [hunk 3]
```diff
@@ -842,6 +842,30 @@ Q: Should a driver support multiple credentials?
...
+    digested and any Unicode character could now appear in a password.
+    Therefore, the SCRAM-SHA-256 mechanism requires passwords to be normalized
+    with SASLprep, in accordance with the SCRAM RFC.
+
+    However, usernames must be unique, which creates a similar upgrade path
+    problem.  SASLprep maps multiple byte representations to a single
+    normalized one.  An existing database could have multiple existing users
+    that map to the same SASLprep form, which makes it impossible to find the
+    correct user document for SCRAM authentication given only a SASLprep
+    username.  After considering various options to address or workaround this
```
Comment (ajdavis) [inline source/auth/auth.rst]: "work around", two words when used as a phrasal verb.

**Pull request closed without merging**